src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
-'use strict';
+"use strict";
+
+const DEFAULT_HEALTH = 100;
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = DEFAULT_HEALTH;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden !== true) {
+      victim.health -= 50;
```

Comment:
I understand now, that 50 is magic number, maybe, but I'm not sure. I made this mistake too